app/questionnaire/questionnaire_manager.py

```diff
@@ -25,40 +26,31 @@ class QuestionnaireManager(object):
     """
     This class represents a user journey through a survey. It models the request/response process of the web application
     """
-    def __init__(self, navigator, schema, json=None):
+    def __init__(self, schema, json=None):
         self._json = json
         self._schema = schema
         self.state = None
 
-        self.navigator = navigator
-
     def validate(self, location, post_data):
 
-        answers = get_answers(current_user)
-
-        if location in self.navigator.get_location_path(answers):
-
-            self.build_state(location, post_data)
+        self.build_state(location, post_data)
 
-            if self.state:
-                # Todo, this doesn't feel right, validation is casting the user values to their type.
-                return self.state.schema_item.validate(self.state)
-            else:
-                # Item has node, but is not in schema: must be introduction, thank you or summary
-                return True
+        if self.state:
+            # Todo, this doesn't feel right, validation is casting the user values to their type.
+            return self.state.schema_item.validate(self.state)
         else:
```

Comment:
We should, but I'd suggest not in this pull request. This behaviour matches what it does currently.